pkg/parse/listener_impl.go

```diff
@@ -3150,3 +3147,11 @@ func (s *TreeShapeListener) currentType() *sysl.Type {
 func (s *TreeShapeListener) setCurrentType(type1 *sysl.Type) {
 	s.typemap[s.fieldname[len(s.fieldname)-1]] = type1
 }
+
+func mustUnescape(endpoint string) string {
```

Comment:
MustEscape + MustUnescape

Reply:
there's `url.PathEscape` that doesn't return error